### PR TITLE
[Core] Fix global pointers vector sort

### PR DESCRIPTION
**Pull request by pooyan-dadvand**
This is a fix for the sort used in the unique method of the global pointers vector.

A test is also provided

### CODE REVIEW

File: kratos/includes/global_pointer.h [hunk 1]
```diff
@@ -302,6 +302,30 @@ struct GlobalPointerComparor
...
+    /**
+     * @brief The () operator
+     * @param pDoF1 The first DoF pointer
+     * @param pDoF2 The second DoF pointer
+     */
+    bool operator()(const GlobalPointer<TDataType>& pGp1, const GlobalPointer<TDataType>& pGp2) const
+    {
+#ifdef KRATOS_USING_MPI
+        return (pGp1.GetRank() == pGp2.GetRank()) ? (pGp1.get() < pGp2.get()) : (pGp1.GetRank() < pGp2.GetRank());
+        }
```
Comment (roigcarlo) [inline kratos/includes/global_pointer.h]:
```suggestion
```

**Comment by roigcarlo** [comment]
Also `comparor` in https://github.com/KratosMultiphysics/Kratos/blob/02b72d8dbac16a47777f8eb75a516a98c081f92f/kratos/includes/global_pointer.h#L288 should be deleted. 

Otherwise we will have two sorting methods. 

We could also use the operators for this...

**Review by roigcarlo — APPROVED** [review]
Also approving because this is urgent.

**Comment by pooyan-dadvand** [comment]
Seems that StructuralMechanicsApplication python tests are failing in travis. @KratosMultiphysics/structural-mechanics

**Comment by philbucher** [comment]
> Seems that StructuralMechanicsApplication python tests are failing in travis. @KratosMultiphysics/structural-mechanics

you can still merge, Travis is no longer required

TBH I am out of ideas
I ran the tests so many times with `GLIB...` with different compilers over and over but I cannot reproduce the occassional CI errors :tipping_hand_man:

**Comment by roigcarlo** [comment]
It would be a shame if someone removes travis from the checklist¿? (and appveyor)

**Comment by pooyan-dadvand** [comment]
I restored the branch because  @roigcarlo may want it for release. @roigcarlo you may delete it after cherry-picking it into release.

**Comment by roigcarlo** [comment]
Na its fine, if its just 2 commits no need for the branch thx!!

**Comment by ddiezrod** [comment]
`SPRErrorProcess1` This test is failing randomly in my Windows machine with FullDebug.  It complains when it tries to do the GetValue in line https://github.com/KratosMultiphysics/Kratos/blob/214850ea3f4e8b807086c739b391fae89c8a5610/applications/StructuralMechanicsApplication/custom_processes/spr_error_process.cpp#L108

Error says "vector iterators in range are from different containers". It seems it is the typical OMP problem where a thread is writing when another one is reading. The test runs fine when using only one thread.

@KratosMultiphysics/structural-mechanics 

I hope nobody will make fun of VisualStudio after this :P

**Comment by philbucher** [comment]
Thanks @ddiezrod !
Can you please take a look @loumalouomega ?

**Comment by loumalouomega** [comment]
OK

**Pull request merged**